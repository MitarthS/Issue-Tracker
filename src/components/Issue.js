import React from 'react';
import { Route, Link , Redirect} from 'react-router-dom'
import IssueForm from './IssueForm';
import {Row, Col} from 'react-bootstrap';
import { isLoggedin } from '../Users/SignInPage';
import issues from '../issues';
import IssueActions from '../actions/IssueActions';
export default class Issue extends React.Component {
    constructor(props){
        super(props);
        this.onEdit = this.onEdit.bind(this);
        this.onDelete = this.onDelete.bind(this);
        
    }
    onEdit(event){
        event.preventDefault();
        //if (isLoggedin)
            var blah = {
                id : this.props.id,
                description : this.props.description,
                severity : this.props.severity,
                status : this.props.status
            }
            //console.log(`${blah.description}`);
            //this.props.onDelete(this.props.id);
            
            //this.props.history.push({pathname : `/editIssue/${this.props.id}`,state : {blah}});
            this.props.history.push(`/editIssue/${this.props.id}`);
    
        //else window.alert(`You must be signed in to edit comments`);
    }
    
    onDelete(event) {
        event.preventDefault();
        console.log(`Entered onDelete inside Issue.js`);
        //if (isLoggedin) 
        this.props.onDelete(this.props.id);
        
        //else window.alert(`You must be signed in to delete comments`);

        //<Col><a href="about:blank" onClick={this.onEdit}>Edit</a></Col>
    }

    render() {
        const path = `/issuedetail/${this.props.description}`;
        const editPath = `/editIssue/${this.props.id}`;
        return (
            <Row>
                <Col>{this.props.id}</Col>
                <Col xs={5}><Link to={path}>{this.props.description}</Link></Col>
                <Col>{this.props.severity}</Col>
                <Col>{this.props.status}</Col>
                <Col><a href="about:blank" onClick={this.onDelete}>Delete</a></Col>
                <Col><Link to={{pathname : `/editIssue/${this.props.id}`,state : {iss : this.props}}}>Edit</Link></Col>
            </Row>
        );
        
    }
}