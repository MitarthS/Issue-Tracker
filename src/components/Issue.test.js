import React from 'react';
import {shallow} from 'enzyme';
import Issue from './Issue';

describe('When no props is passed to Issue ', () => {
    it('renders correctly', () => {
        let wrapper = shallow(<Issue/>);
        expect(wrapper).toMatchSnapshot();
    });
});

describe('When issues array is passed to Issue', () => {
    let wrapper;
    let props;

    beforeEach(() => {
        props = {
            issues: [
                {
                    "id":1,
                    "description": "Raj",
                    "serverity": "Critical",
                    "status": "Open"                    
                },
                {
                    "id":2,
                    "description": "Ramesh",
                    "serverity": "Critical",
                    "status": "Open"
                }
            ]
        }
        wrapper = shallow(<Issue {...props}/>);
    });

    it('renders correctly when passed the  props', () => {
        expect(wrapper).toMatchSnapshot();
    });

});

describe('When issues array passed to Issue is null', () => {
    let wrapper;
    let props;

    beforeEach(() => {
        props = {
            issues: null
        }
        wrapper = shallow(<Issue {...props}/>);
    });

    it('should not crash', () => {
        expect(wrapper).toMatchSnapshot();
    });

});
