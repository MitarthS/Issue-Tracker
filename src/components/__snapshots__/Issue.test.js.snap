// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When issues array is passed to Issue renders correctly when passed the  props 1`] = `ShallowWrapper {}`;

exports[`When issues array passed to Issue is null should not crash 1`] = `ShallowWrapper {}`;

exports[`When no props is passed to Issue  renders correctly 1`] = `ShallowWrapper {}`;
